pkg/api/validate/openshiftcluster_validatedynamic.go

```diff
@@ -287,3 +294,26 @@ func (dv *openShiftClusterDynamicValidator) validateVnet(ctx context.Context, vn
 
 	return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidLinkedVNet, "", "The provided vnet '%s' is invalid: custom DNS servers are not supported.", vnetID)
 }
+
+func (dv *openShiftClusterDynamicValidator) validateProviders(ctx context.Context, providerClient resources.ProvidersClient) error {
+	providers := []string{
+		"Microsoft.Storage",
+		"Microsoft.Authorization",
+		"Microsoft.Compute",
+		"Microsoft.Network",
+	}
+
+	providersList, err := providerClient.List(ctx, nil, "")
+	if err != nil {
+		return err
+	}
+	for _, rp := range providersList {
+		for _, provider := range providers {
```

Comment:
this will not detect the case where the provider of interest is not returned in providersList, and it should do.